tools/codegen/gen_functionalization_type.py

```diff
@@ -195,7 +195,22 @@ def emit_inplace_functionalization_body(
                     if arg.type.is_tensor_like() else None,
         f.func.arguments.flat_all))
 
-    if functional_op is None:
+    # Note [functionalizating copy_() and not preserving strides]
+    # copy_() can't be functionalized, since there doesn't exist an out-of-place variant.
+    # We could add one, but that would be sub-optimal for functorch: copy() would need to allocate a fresh tensor.
+    # This may seem like a large hack for one optimization, but copy_() is one of the most common inplace operators.
+    # Instead, we can replace `self.copy_(src)` with `src.to(self).expand_as(self)`.
+    # This maintains the exact same semantics, EXCEPT that we don't preserve the strides from `self`.
```

Comment:
sidenote: I think we already knew this, but in general functionalization doesn't preserve the "same semantics" as the original function. This is probably OK but we might need to figure out how to better explain that to whoever uses it. Here's an example:
```
def foo(x, y):
  return x.add_(y)

x = torch.randn(1)
y = torch.randn(3)

# Produces an error in normal PyTorch
foo(x, y)

# Inside functionalization, x.add_(y) becomes x + y, and no error is produced!
# Also: I guess this means that torch/xla has slightly different semantics from eager-mode PyTorch as well?
functionalize(foo)(x, y)
```

Reply:
Yep, good point, and these differences probably mostly match up with pytorch/xla's differences from cpu eager mode (although pytorch/xla has a few more differences due to bugs, that will get fixed by the pass).